scripts/2.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,535 @@
+
+# coding: utf-8
+
+# # Predicting Aberrations in The Hippo Signaling Pathway
+
+# ## To do this, we will use Hetnets to query "Signaling by Hippo"
+# 
+# The query is built in the [Cypher Language](https://neo4j.com/developer/cypher-query-language/) and draws data from [Hetionet](https://neo4j.het.io/browser/)
+# 
+# ### How Cognoma could help with Hippo signaling
+# 
+# The Hippo pathway is a highly conserved signaling cascade that controls organ size, cell growth, and cell death ([Zhao et al. 2010](http://doi.org/10.1101/gad.1909210)). It is one of the mechanisms that influences size diversity across eukaryotes; including different sizes across dog breeds ([Dong et al. 2007](http://doi.org/10.1016/j.cell.2007.07.019), [Crickmore and Mann 2008](http://doi.org/10.1002/bies.20806)). Recently, Hippo signaling has also been shown to be important for tumorigenesis, but there are shockingly few recurrent mutations of single genes within the pathway across tissues ([Harvey et al 2013](http://doi.org/10.1038/nrc3458)). Therefore, leveraging cancers from multiple tissues and combining genes associated with the same pathway could aid in the detection of a Hippo signaling specific gene expression signature. Cognoma is situated well to quickly query the list of all pathway associated genes, build a machine learning classifier to detect aberrant pathway activity, and output tissue and gene specific performance.
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+from neo4j.v1 import GraphDatabase
+
+
+# In[3]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[4]:
+
+REACTOMEPATHWAY = 'PC7_7459'  # "Signaling by Hippo" in Reactome
```

Comment:
Just in case, does anyone else think the identifier we are using here does not match what is currently on Reactome? http://www.reactome.org/content/detail/R-HSA-2028269
Of course we are querying the hetnet. But when I type that ID into Reactome, I don't get anything. Whereas typing the description yields the link above, with a seemingly different ID. I only mention this in case there are potential downstream issues, like our issues with Entrez IDs in the cancer-data group. I may simply be misusing Reactome and this probably doesn't matter, so apologies in advance if this is the case.